File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/SerializableClasses/Precancel/BookingItineraryResponseRoom.vb
Imports Newtonsoft.Json
Imports ThirdParty.VBSuppliers.ExpediaRapid.SerializableClasses.Book

Namespace ExpediaRapid.SerializableClasses.BookingItinerary
    Public Class BookingItineraryResponseRoom
        Inherits BookRequestRoom

        <JsonProperty("id")>
        Public Property RoomID As String

        <JsonProperty("confirmation_id")>
        Public Property ConfirmationID As ConfirmationID

        <JsonProperty("bed_group_id")>
        Public Property BedGroupID As String

        <JsonProperty("checkin")>
        Public Property CheckIn As Date

        <JsonProperty("checkout")>
        Public Property CheckOut As Date

        <JsonProperty("number_of_adults")>
        Public Property NumberOfAdults As Integer

        <JsonProperty("child_ages")>
        Public Property ChildAges As New List(Of Integer)

        <JsonProperty("status")>
        Public Property Status As String

        <JsonProperty("rate")>
        Public Property Rate As BookingItineraryResponseRoomRate

        <JsonProperty("links")>
        Public Property Links As New Dictionary(Of String, Link)

    End Class

End Namespace

File: src/libs/iVectorOne.Suppliers/ExpediaRapid/SerializableClasses/Precancel/BookingItineraryResponseRoom.cs
namespace iVectorOne.Suppliers.ExpediaRapid.SerializableClasses.BookingItinerary
{
    using System;
    using System.Collections.Generic;
    using Newtonsoft.Json;
    using iVectorOne.Suppliers.ExpediaRapid.SerializableClasses.Book;

    public class BookingItineraryResponseRoom : BookRequestRoom
    {

        [JsonProperty("id")]
        public string RoomID { get; set; }

        [JsonProperty("confirmation_id")]
        public ConfirmationID ConfirmationID { get; set; }

        [JsonProperty("bed_group_id")]
        public string BedGroupID { get; set; }

        [JsonProperty("checkin")]
        public DateTime CheckIn { get; set; }

        [JsonProperty("checkout")]
        public DateTime CheckOut { get; set; }

        [JsonProperty("number_of_adults")]
        public int NumberOfAdults { get; set; }

        [JsonProperty("child_ages")]
        public List<int> ChildAges { get; set; } = new List<int>();

        [JsonProperty("status")]
        public string Status { get; set; }

        [JsonProperty("rate")]
        public BookingItineraryResponseRoomRate Rate { get; set; }

        [JsonProperty("links")]
        public Dictionary<string, Link> Links { get; set; } = new Dictionary<string, Link>();

    }

}